clang/lib/Headers/hlsl/hlsl_compat_overloads.h

```diff
@@ -277,6 +277,12 @@ constexpr bool4 isinf(double4 V) { return isinf((float4)V); }
 // lerp builtins overloads
 //===----------------------------------------------------------------------===//
 
+template <typename T, uint N>
+constexpr __detail::enable_if_t<(N > 1 && N <= 4), vector<T, N>>
```

Comment:
I don't think compat overload is the right place for this. Seems like we might want to keep this around past 202x.

Reply:
I also think it feels a bit odd, but I was following the examples from [clamp](https://github.com/llvm/llvm-project/issues#128230) and [min/max](https://github.com/llvm/llvm-project/issues#128231). I think if we're going to move them, we should do all of these together.